kuwala/core/canvas/src/components/Modals/TransformationCatalog/TransformationCatalogModal.js

```diff
@@ -0,0 +1,329 @@
+                    <span className={'font-semibold'}>
+                        {optionItem.name}
+                    </span>
+                </div>
+            </Button>
+        )
+    }
+
+    const renderSelectedOptionDetails = () => {
+        if(selectedCatalogOption === null){
+            return (
+                <div className={Classes.OptionDetailsAlertText}>
+                    <p>To see the <span className={'text-kuwala-purple'}>details</span> of a transformation block,</p>
+                    <p>select one from the left</p>
+                </div>
+            )
+        } else {
+            const optionItem = catalogOptions[selectedCatalogOption];
+            return (
+                <div className={Classes.OptionDetailsContent}>
+                    <div className={'text-kuwala-purple space-x-4'}>
+                        {getCatalogItemIcon(optionItem.icon)}
+                        <span className={'font-semibold'}>
+                            {optionItem.name}
+                        </span>
+                    </div>
+                    <p className={'mt-2'}>
+                        {optionItem.description}
+                    </p>
+                    <div className={cn(Classes.RequiredColumnBase, 'mt-6')}>
+                        <span>
+                            Required column types:
+                        </span>
+                        {renderBadgeList(optionItem.required_column_types)}
+                    </div>
+                    <div className={cn(Classes.RequiredColumnBase, 'mt-2')}>
+                        <span>
+                            Optional column types:
+                        </span>
+                        {renderBadgeList(optionItem.optional_column_types)}
+                    </div>
+
+                    <div className={Classes.OptionDetailsParameterAndExample}>
+
+                        <div className={Classes.OptionDetailsParameterContainer}>
+                            <p>Parameters</p>
+                            {optionItem.macro_parameters.map((el) => <li>{el.name}</li>)}
+                        </div>
+
+                        <div className={Classes.OptionDetailsExampleContainer}>
+                            {renderExampleTableWrapper(optionItem)}
+                        </div>
+
+                    </div>
+
+                </div>
+            )
+        }
+    }
+
+    const renderExampleTableWrapper = (optionItem) => {
+        return (
+            <>
+                {optionItem.examples_before.map((el) =>
+                    renderExampleTable({
+                        examplesData: el,
+                        text: 'Before'
+                    })
+                )}
+                {optionItem.examples_after.map((el) =>
+                    renderExampleTable({
+                        examplesData: el,
+                        text: 'After'
+                    })
+                )}
+            </>
+        )
+    }
```

Comment:
The heading `Before` and `After` should only be written once. If there are multiple tables in one section (see screenshot) they should be rendered next to each other in one row.

![image](https://user-images.githubusercontent.com/28758040/165485114-b5c7d9ff-db79-4b0e-b02c-f40d56b27b6e.png)

Reply:
done, also refactored into smaller functions.